### PR TITLE
Add signinWithTwitter

### PR DESCRIPTION
**Pull request by azihsoyn**
Add twitter signin for firebase_auth.

### CODE REVIEW

File: packages/firebase_auth/lib/firebase_auth.dart [hunk 1]
```diff
@@ -215,6 +215,21 @@ class FirebaseAuth {
     return currentUser;
   }
 
+  Future<FirebaseUser> signInWithTwitter({
```
Comment (collinjackson) [inline packages/firebase_auth/lib/firebase_auth.dart]: We should add Dartdoc for this (along with the other methods if you're feeling ambitious)

File: packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java [hunk 2]
```diff
@@ -235,6 +238,17 @@ private void handleSignInWithFacebook(MethodCall call, final Result result) {
         .addOnCompleteListener(new SignInCompleteListener(result));
   }
 
+  private void handleSignInWithTwitter(MethodCall call, final Result result) {
+    @SuppressWarnings("unchecked")
+    Map<String, String> arguments = (Map<String, String>) call.arguments;
+    String authToken = arguments.get("authToken");
+    String authTokenSecret = arguments.get("authTokenSecret");
```
[the remark this answers is not in the document]
Comment (azihsoyn) [reply packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java]: Thanks, advise! I'll correct it.

**Comment by azihsoyn** [comment]
Is there anything else to do?

File: packages/firebase_auth/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.5.9
+
+* Added support to signinWithTwitter
```
Comment (mravn-google) [inline packages/firebase_auth/CHANGELOG.md]: The change log should probably be more prose and less code:
```
* Added support for signing in with a Twitter account.
```

File: packages/firebase_auth/lib/firebase_auth.dart [hunk 4]
```diff
@@ -220,6 +220,26 @@ class FirebaseAuth {
     return currentUser;
   }
 
+  /// Creates new user with twitter account and returns [Future<FirebaseUser>]
+  ///
+  /// throws [PlatformException] when
+  /// 1. twitter email address is already used
+  /// Detailed documentation on possible error causes can be found in [Android docs](https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseUser#exceptions_5) and [iOS docs](https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRAuth#/c:objc(cs)FIRAuth(im)signInWithCredential:completion:)
+  Future<FirebaseUser> signInWithTwitter({
```
Comment (mravn-google) [inline packages/firebase_auth/lib/firebase_auth.dart]: I suggest to keep this shorter here, e.g. just
```dart
/// Signs in with a Twitter account using the specified credentials.
///
/// The returned future completes with the signed-in user or a [PlatformException], if sign in failed.
```

**Comment by mravn-google** [comment]
Published.

**Comment by azihsoyn** [comment]
Great thanks reviewing and merging!

**Pull request merged**